src/Mono.Android/Xamarin.Android.Net/AndroidClientHandler.cs

```diff
@@ -177,23 +184,36 @@ protected void AssertSelf ()
 			if (!request.RequestUri.IsAbsoluteUri)
 				throw new ArgumentException ("Must represent an absolute URI", "request");
 
-			URL java_url = new URL (request.RequestUri.ToString ());
-			URLConnection java_connection = java_url.OpenConnection ();
-			HttpURLConnection httpConnection = await SetupRequestInternal (request, java_connection);
-			return await ProcessRequest (request, java_url, httpConnection, cancellationToken);
+			var redirectState = new RequestRedirectionState {
+				NewUrl = request.RequestUri,
+				RedirectCounter = 0,
+				Method = request.Method
+			};
+			while (true) {
+				URL java_url = new URL (redirectState.NewUrl.ToString ());
```

Comment:
By any chance can we wrap `java_url` and `java_connection` in `using` blocks? Their lifetime *looks* limited, and doing so would help control the GREF count.